app/src/features/rbac/utils.ts

```
@@ -0,0 +1,95 @@
+import Logger from "lib/logger";
+import { ROLE_TO_RESOURCE_PERMISSIONS_MAP } from "./constants/config";
+import { RBAC } from "./types";
+import { toast } from "utils/Toast";
+
+export const validateResourcePermissionByRole = (
+  role: RBAC.Role,
+  resource: RBAC.Resource,
+  permissionToValidate: RBAC.Permission
+):
+  | {
+      isValid: true;
+    }
+  | { isValid: false; error: RBAC.Error } => {
+  try {
+    if (!(role in RBAC.Role)) {
+      return {
+        isValid: false,
+        error: RBAC.Error.invalid_role,
+      };
+    }
+
+    if (!(resource in RBAC.Resource)) {
+      return {
+        isValid: false,
+        error: RBAC.Error.invalid_resource,
+      };
+    }
+
+    if (!(permissionToValidate in RBAC.Permission)) {
+      return {
+        isValid: false,
+        error: RBAC.Error.invalid_permission,
+      };
+    }
+
+    const resourcePermissions = ROLE_TO_RESOURCE_PERMISSIONS_MAP[role][resource];
+
+    const isAllowed = resourcePermissions[permissionToValidate];
+
+    if (!isAllowed) {
+      return {
+        isValid: false,
+        error: RBAC.Error.not_allowed,
+      };
+    }
+
+    return { isValid: true };
+  } catch (error) {
+    return {
```

Remove `try/catch`, a genuine error will be parsed as `invalid_permission`. If any error happens, let it bubble up.